lib/src/commands/sublist.dart

```diff
@@ -3,19 +3,42 @@ part of dslink.dql.query;
 class SublistHolder {
   final String path;
 
+  QueryUpdate oldUpdate;
+
   SublistHolder(this.path);
 
   SublistListQueryProcessor listProcessor;
+  StreamController<QueryUpdate> controller;
+
   StreamSubscription<QueryUpdate> listSub;
-  List<String> _handles = <String>[];
+  Map<String, QueryUpdate> _handles = <String, QueryUpdate>{};
 
   Iterable<String> destroy() {
     if (listSub != null) {
       listSub.cancel();
       listSub = null;
     }
 
-    return _handles;
+    if (controller != null) {
+      controller.close();
+      controller = null;
+    }
+
+    return _handles.keys;
+  }
+
+  void onOldUpdate() {
```

Comment:
I'm not sure I understand what's the deal with the "oldUpdate" thing.. Care to explain?

Reply:
We have a stream of table updates, and we are interjecting that update into new updates that match.

```
path | something
abc   | 123
```

=>

```
path | something
abc/1 | 123
abc/2 | 123
abc/3 | 123
```